sentry_openproject/plugin.py

```diff
@@ -110,45 +83,208 @@ def _get_group_description(self, request, group, event):
+            'default': '',
+            'type': 'select',
+            'has_autocomplete': True,
+        }, {
+            'name': 'comment',
+            'label': 'Comment',
+            'default': absolute_uri(group.get_absolute_url()),
+            'type': 'textarea',
+            'help': ('Leave blank if you don\'t want to '
+                     'add a comment to the OpenProject work package.'),
+            'required': False
+        }]
+
+    def get_issue_label(self, group, issue_id, **kwargs):
+        return 'WP#%s' % issue_id
+
+    def get_issue_url(self, group, issue_id, **kwargs):
+        url = self.get_option('url', group.project)
+        return urlparse.urljoin(url, '/work_packages/{}'.format(issue_id))
+
+    def get_client(self, project, user):
+        url = self.get_option('url', project)
+        apikey = self.get_option('apikey', project)
+        if not all((url, apikey)):
+            raise PluginError('OpenProject plugin not correctly configured!')
+        return OpenProjectClient(url, apikey)
+
+    def create_issue(self, request, group, form_data, **kwargs):
+        client = self.get_client(group.project, request.user)
+        try:
+            response = client.create_work_package(
+                self.get_option('project_slug', group.project),
+                form_data['title'],
+                form_data['type'],
+                description=form_data.get('description'),
+                assignee_id=form_data.get('assignee'),
+            )
+        except Exception as e:
+            self.raise_error(e)
+
+        return response['id']
+
+    def link_issue(self, request, group, form_data, **kwargs):
+        client = self.get_client(group.project, request.user)
+        try:
+            issue = client.get_work_package(form_data['issue_id'])
+        except Exception as e:
+            self.raise_error(e)
+
+        comment = form_data.get('comment')
+        if comment:
+            try:
+                client.create_comment(issue['id'], comment)
+            except Exception as e:
+                self.raise_error(e)
+
+        return {
+            'title': issue['subject'],
         }
-        if assignee:
-            payload['_links'].update({
-                'assignee': {
-                    'href': '/api/%s/users/%s' % (API_VERSION, assignee)
-                }
-            })
 
-        session = http.build_session()
-        r = session.post(url, data=json.dumps(payload), headers=headers)
-        data = json.loads(r.text)
+    def get_configure_plugin_fields(self, request, project, **kwargs):
+        apikey = self.get_option('apikey', project)
+        apikey_field = get_secret_field_config(
+            apikey,
+            'This is API key of an OpenProject user, who will be the author '
+            'of the issue on OpenProject.',
+            include_prefix=True,
+        )
+        apikey_field.update({
+            'name': 'apikey',
+            'label': 'OpenProject API key',
```

Comment:
Yeah, borrowed the idea from original [sentry plugins](https://github.com/getsentry/sentry-plugins/blob/master/src/sentry_plugins/jira/plugin.py#L507). Also, once set, the key is not displayed in the form.